pallets/pips/src/lib.rs

```diff
@@ -528,95 +568,126 @@ decl_module! {
         }
 
         /// A network member creates a PIP by submitting a dispatchable which
         /// changes the network in someway. A minimum deposit is required to open a new proposal.
         ///
         /// # Arguments
+        /// * `proposer` is either a signing key or committee.
+        ///    Used to understand whether this is a committee proposal and verified against `origin`.
         /// * `proposal` a dispatchable call
-        /// * `deposit` minimum deposit value
+        /// * `deposit` minimum deposit value, which is ignored if `proposer` is a committee.
         /// * `url` a link to a website for proposal discussion
         #[weight = (1_850_000_000, DispatchClass::Operational, Pays::Yes)]
         pub fn propose(
             origin,
+            proposer: Proposer<T::AccountId>,
             proposal: Box<T::Proposal>,
             deposit: BalanceOf<T>,
             url: Option<Url>,
             description: Option<PipDescription>,
-            beneficiaries: Option<Vec<Beneficiary<T::Balance>>>
         ) -> DispatchResult {
-            let proposer = ensure_signed(origin)?;
-
-            // Pre conditions: caller must have min balance
-            ensure!(
-                deposit >= Self::min_proposal_deposit(),
-                Error::<T>::IncorrectDeposit
-            );
+            // 1. Ensure it's really the `proposer`.
+            Self::ensure_signed_by(origin, &proposer)?;
+
+            // 2. Add a deposit for community PIPs.
+            if let Proposer::Community(ref proposer) = proposer {
+                // ...but first make sure active PIP limit isn't crossed.
+                // This doesn't apply to committee PIPs.
+                // `0` is special and denotes no limit.
+                let limit = ActivePipLimit::get();
+                ensure!(limit == 0 || ActivePipCount::get() < limit, Error::<T>::TooManyActivePips);
+
+                // Pre conditions: caller must have min balance.
+                ensure!(deposit >= Self::min_proposal_deposit(), Error::<T>::IncorrectDeposit);
+
+                // Reserve the minimum deposit.
+                <T as Trait>::Currency::reserve(&proposer, deposit)
+                    .map_err(|_| Error::<T>::InsufficientDeposit)?;
+            }
 
-            // Reserve the minimum deposit
-            <T as Trait>::Currency::reserve(&proposer, deposit).map_err(|_| Error::<T>::InsufficientDeposit)?;
+            // 3. Charge protocol fees, even for committee PIPs.
             <T as IdentityTrait>::ProtocolFee::charge_fee(ProtocolOp::PipsPropose)?;
 
+            // 4. Construct and add PIP to storage.
             let id = Self::next_pip_id();
-            let curr_block_number = <system::Module<T>>::block_number();
-            let cool_off_until = curr_block_number + Self::proposal_cool_off_period();
-            let end = cool_off_until + Self::proposal_duration();
+            ActivePipCount::mutate(|count| *count += 1);
```

Comment:
I wonder if we should make this count just for community PIPs - otherwise committees can fill up the limit and stifle community Pips.

Reply:
Should be easy to fix, but note that committee PIPs still add computational complexity to snapshot-taking, so excluding them from the count will not account for that complexity.

Reply:
Agree - lets leave as is.